ros2param/ros2param/api/__init__.py

```diff
@@ -135,103 +63,34 @@ def load_parameter_dict(*, node, node_name, parameter_dict):
-        if node_name in param_file:
-            param_keys.append(node_name)
-
-        if param_keys == []:
-            raise RuntimeError('Param file does not contain parameters for {}, '
-                               ' only for nodes: {}' .format(node_name, param_file.keys()))
-        param_dict = {}
-        for k in param_keys:
-            value = param_file[k]
-            if type(value) != dict or 'ros__parameters' not in value:
-                raise RuntimeError('Invalid structure of parameter file for node {}'
-                                   'expected same format as provided by ros2 param dump'
-                                   .format(k))
-            param_dict.update(value['ros__parameters'])
-        load_parameter_dict(node=node, node_name=node_name, parameter_dict=param_dict)
-
-
 def call_describe_parameters(*, node, node_name, parameter_names=None):
-    # create client
-    client = node.create_client(
-        DescribeParameters, f'{node_name}/describe_parameters')
-
-    # call as soon as ready
-    ready = client.wait_for_service(timeout_sec=5.0)
-    if not ready:
-        raise RuntimeError('Wait for service timed out')
-
-    request = DescribeParameters.Request()
-    if parameter_names:
-        request.names = parameter_names
-    future = client.call_async(request)
+    client = AsyncParameterClient(node, node_name)
+    future = client.describe_parameters(parameter_names)
     rclpy.spin_until_future_complete(node, future)
-
-    # handle response
     response = future.result()
     return response
 
 
 def call_get_parameters(*, node, node_name, parameter_names):
-    # create client
-    client = node.create_client(GetParameters, f'{node_name}/get_parameters')
-
-    # call as soon as ready
-    ready = client.wait_for_service(timeout_sec=5.0)
-    if not ready:
-        raise RuntimeError('Wait for service timed out')
-
-    request = GetParameters.Request()
-    request.names = parameter_names
-    future = client.call_async(request)
+    client = AsyncParameterClient(node, node_name)
+    future = client.get_parameters(parameter_names)
     rclpy.spin_until_future_complete(node, future)
-
-    # handle response
     response = future.result()
     return response
 
 
 def call_set_parameters(*, node, node_name, parameters):
-    # create client
-    client = node.create_client(SetParameters, f'{node_name}/set_parameters')
-
-    # call as soon as ready
-    ready = client.wait_for_service(timeout_sec=5.0)
-    if not ready:
-        raise RuntimeError('Wait for service timed out')
-
-    request = SetParameters.Request()
-    request.parameters = parameters
-    future = client.call_async(request)
+    client = AsyncParameterClient(node, node_name)
+    future = client.set_parameters(parameters)
     rclpy.spin_until_future_complete(node, future)
-
-    # handle response
     response = future.result()
     return response
 
 
 def call_list_parameters(*, node, node_name, prefix=None):
```

Comment:
I noticed we're not using (nor were we previously using) the `prefix` argument. I suppose we could just remove it:

```suggestion
def call_list_parameters(*, node, node_name):
```

Reply:
It's used in https://github.com/ros2/ros2cli/blob/4d6463b7442d8f5b7945c92182826d12acd7bdc3/ros2param/ros2param/verb/list.py#L76

Reply:
Oops :upside_down_face: